samples/parselib/src/main/cpp/wav/WavStreamReader.h

```diff
@@ -59,13 +59,13 @@ class WavStreamReader {
 protected:
     InputStream *mStream;
 
-    WavRIFFChunkHeader *mWavChunk;
-    WavFmtChunkHeader *mFmtChunk;
-    WavChunkHeader *mDataChunk;
+    std::shared_ptr<WavRIFFChunkHeader> mWavChunk;
+    std::shared_ptr<WavFmtChunkHeader> mFmtChunk;
+    std::shared_ptr<WavChunkHeader> mDataChunk;
 
     long mAudioDataStartPos;
 
-    std::map<RiffID, WavChunkHeader *> *mChunkMap;
+    std::map<RiffID, std::shared_ptr<WavChunkHeader>> *mChunkMap;

```

Comment:
I don't see where this map is used. Can we remove it?
Then we could use std::unique_ptr or, better yet, just declare the Chunks and not use new.

Reply:
It's there so that the WavStreamReader can evolve and also chunks that it doesn't recognize. I'd like to leave it and see how WavStreamReader evolves moving forward.